jiant/modules/span_modules.py

```diff
@@ -78,6 +80,7 @@ def forward(
         sent_mask: torch.Tensor,
         task: Task,
         predict: bool,
+        cuda_device: int,
```

Comment:
Is there a way to have this live with trainer or somewhere else external, to avoid having to pass it around?

Reply:
:+1:, or at least live with the top level model. Alternatively, is there a way to access the device from `self`, e.g. `self.device`?

I don't totally understand why you need to move `n_exs` to device

Reply:
Yep, I'll move it up. 
As to your question, what data parallel does is that it makes copies of the model and then sends subsets of the batches to each of these copies (in each of the GPUs), then aggregates all the out dictionaries of all of these copies in one GPU. Thus, everything in the out dictionary needs to be a tensor in cuda.